### ISSUE TITLE
Get rid of ButtonModel.isA11yClicking()?

### ISSUE THREAD

**Issue by zepumph** [phetsims/sun#639]
@pixelzoom brought this up.

We, in other common code, have been solidifying a pattern for knowing if a listener event comes from the PDOM or elsewhere. Ideally it looks like this (checking on the event):

`SceneryEvent,isFromPDOM()`

From this and I feel like ButtonModel.isA11yClicking is an old, vestigial pattern that we may not need anymore. 

@pixelzoom also noted that all usages of it breaks buttonModel's visibility annotation by reaching into the model from a button instance, like so:

```

      listener: () => {
        options.closeButtonListener();

        // if listener was fired because of accessibility
        if ( closeButton.buttonModel.isA11yClicking() ) {
          this.focusOnCloseNode && this.focusOnCloseNode.focus();
        }
      },
```

If we no longer have isA11yClicking(), then we can likely get rid of `isA11yClickingProperty`, which is what that function ran on. 

@jessegreenberg can we discuss tomorrow at our meeting?

**Comment by zepumph** [comment]
![image](https://user-images.githubusercontent.com/6856943/96489351-d1379680-11eb-11eb-9796-09cb44fa62be.png)

Things have been like this through the last ~10 sim publications (including those with interactive description). No logic issues have been reported in QA testing or elsewhere. Is this in response to the face that this pattern accesses a `protected` member from outside the instance?

**Comment by pixelzoom** [comment]
> Is this in response to the face that this pattern accesses a protected member from outside the instance?

Yes - violating an API is a bug. And there are now 4 call sites that rely on that programming error. The only reason that it doesn't manifest as a hard failure is JavaScript's lack of language support for visibility.

**Comment by pixelzoom** [comment]
I will reluctantly remove the "bug" label here, since it is not currently causing any unintended behavior. Perhaps guess we need a "programmer error" label for cases like this.

**Comment by jessegreenberg** [comment]
I am not sure if we can get rid of this, it will depend on whether we can send the SceneryEvent to listeners on the button. Many button models fire from ButtonModel.downProperty separate from the SceneryEvent. I am not seeing a clear way to replace it yet, but definitely something we can look at tomorrow.

**Comment by zepumph** [comment]
@jessegreenberg and I talked about this today. We looked into trying to use the preferred pattern for this, which is `SceneryEvent.isFromPDOM` but found that we couldn't. This is a limitation with sun buttons because of how the scenery listener events are obfuscated when as they control ButtonModel Properties like `downProperty`. Thus, when the button models cause firing of sun button listeners based on listeners to the `downProperty`, we don't have access to the underlying scenery event.

While this is a bit of a bummer, we weren't too worried about tackling this in a bit different of a way. Above we refactored the pattern to use a `isPDOMClicking()` function on the button Nodes. This function ties directly into the `PressListener.pdomClickingProperty`. We like this a lot because it feels more MVC (one of the few times in which the MVC of sun buttons has made sense to me). The input listener is added to the Node, and so it should be the Node that determines where that input is coming from. The model shouldn't care about how the button was pressed, it should only care that it is now in the down state. 

Furthermore, this means that we were able to remove the `a11yClickingProperty` from ButtonModel. Yay!

There was a minor duplication added to make `isPDOMClicking` a function in JoistButton and in ButtonNode (it is likely that this will be removed by refactoring JoistButton to extend ButtonNode in the future), but other that this was a simple improvement. 

@pixelzoom will you please review this. If you would prefer, feel free to respond to larger sun-button/a11y related thoughts over in https://github.com/phetsims/sun/issues/633. Then we can close this issue.

**Comment by pixelzoom** [comment]
> @pixelzoom will you please review this.

I don't understand this topic at all, so I'll take your word for it.  

>  If you would prefer, feel free to respond to larger sun-button/a11y related thoughts over in #633. 

Have done so.

> Then we can close this issue.

Closing.
